shard/committee/assignment.go

```diff
@@ -116,62 +124,131 @@ func preStakingEnabledCommittee(s shardingconfig.Instance) shard.State {
 	return shardState
 }
 
-func with400Stakers(
-	s shardingconfig.Instance, stakerReader StakingCandidatesReader,
+func eposStakedCommittee(
+	s shardingconfig.Instance, stakerReader DataProvider, stakedSlotsCount int,
 ) (shard.State, error) {
 	// TODO Nervous about this because overtime the list will become quite large
 	candidates := stakerReader.ValidatorCandidates()
-	stakers := make([]*staking.Validator, len(candidates))
+	essentials := map[common.Address]effective.SlotOrder{}
+	slotUsage := map[common.Address]int{}
```

Comment:
probably don't need slotUsage if we return a ordered slots after median and only use the top 320.